app/helpers/application_helper.rb

```diff
@@ -1,17 +1,22 @@
 module ApplicationHelper
    # @return the path to the login page
   def login_url
-    '/auth/cognito-idp/'
+    AUTH_LOGIN_PATH
   end
 
   def logout_url
-    host = "#{request.protocol}#{request.host}:#{request.port}" || "http://localhost:3000"
-    URI::HTTPS.build(
-        host: Rails.application.secrets.cognito_user_pool_site,
-        path: "/logout",
-        query: {
-            "client_id": Rails.application.secrets.cognito_client_id,
-            "logout_uri": "#{host}/logout/callback"
-        }.to_query)
+    if session[:provider] == 'cognito-idp'
+        host = "#{request.protocol}#{request.host}:#{request.port}" || "http://localhost:3000"
+        URI.join(
+            Rails.application.secrets.cognito_user_pool_site,
+            "logout?"+
+            {
+                "client_id": Rails.application.secrets.cognito_client_id,
+                "logout_uri": "#{host}#{logout_callback_path}"
```

Comment:
I think the string interpolation can be replaced with `logout_callback_url`

Reply:
could we move the `{...}.to_query` into a variable so we can do `"logout?#{query_params}"`?